lib/run_migrations_with_concurrent_guard.rb

```diff
@@ -2,6 +2,10 @@
 # for any concurrent failures;
 
 begin
+  # Offset all the different rails c's against each other over 30secs
+  # Put it in this begin so it keeps on happening on each retry.
+  sleep(rand * 30)
```

Comment:
This is unrelated to this PR really, but it just stops us killing the db by having 20 new servers all try and run migrations simultaneously.